web/src/typeahead_helper.ts

```diff
@@ -555,6 +556,153 @@ export function sort_recipients<UserType extends UserOrMentionPillData | UserPil
     return recipients.slice(0, max_num_items);
 }
 
+export function compare_setting_options(
+    option_a: UserPillData | UserGroupPillData,
+    option_b: UserPillData | UserGroupPillData,
+    target_group: UserGroup | undefined,
+): number {
+    if (option_a.type === "user_group" && option_b.type === "user") {
+        return -1;
+    }
+
+    if (option_b.type === "user_group" && option_a.type === "user") {
+        return 1;
+    }
+
+    if (option_a.type === "user_group" && option_b.type === "user_group") {
+        const user_group_a = user_groups.get_user_group_from_id(option_a.id);
+        const user_group_b = user_groups.get_user_group_from_id(option_b.id);
+
+        if (user_group_a.is_system_group && !user_group_b.is_system_group) {
+            return -1;
+        }
+
+        if (user_group_b.is_system_group && !user_group_a.is_system_group) {
+            return 1;
+        }
+
+        if (user_group_a.name < user_group_b.name) {
+            return -1;
+        }
+
+        return 1;
```

Comment:
Technically a sorting function should return 0 if the values are the same, right? But I guess that should not happen.

Reply:
Yes, two groups cannot have same names.